std/range/package.d

```diff
@@ -928,9 +928,17 @@ See_Also: $(LREF only) to chain values to a range
  */
 auto chain(Ranges...)(Ranges rs)
 if (Ranges.length > 0 &&
-    allSatisfy!(isInputRange, staticMap!(Unqual, Ranges)) &&
-    !is(CommonType!(staticMap!(ElementType, staticMap!(Unqual, Ranges))) == void))
+    anySatisfy!(isInputRange, staticMap!(Unqual, Ranges)))
```

Comment:
Is there a reason to have this `anySatisfy` constraint here but not on `chooseAmong`?

Reply:
I've added this constraint to chooseAmong now for consistency. The only reason was I thought that was a less likely name to have a collision (see my next comment).